pvlib/test/test_scaling.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+import pandas as pd
+
+import pytest
+from numpy.testing import assert_almost_equal
+
+from pvlib import scaling
+from conftest import requires_scipy
+
+
+# Sample positions
+lat = np.array((9.99, 10, 10.01))
+lon = np.array((4.99, 5, 5.01))
+coordinates = np.array([(lati, loni) for (lati, loni) in zip(lat, lon)])
+
+# Sample cloud speed
+cloud_speed = 5
+# Generate a sample clear_sky_index and time vector.
+clear_sky_index = np.ones(10000)
+clear_sky_index[5000:5005] = np.array([1, 1, 1.1, 0.9, 1])
+time = np.arange(0, len(clear_sky_index))
+
+# Sample dt
+dt = 1
+
+# Expected positions for sample lat/lon given above (calculated manually)
+expect_xpos = np.array([554863.4, 555975.4, 557087.3])
+expect_ypos = np.array([1110838.8, 1111950.8, 1113062.7])
+
+# Sample positions based on the previous lat/lon
+positions = np.array([pt for pt in zip(expect_xpos, expect_ypos)])
+
+# Expected timescales for dt = 1
+expect_tmscale = [2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096]
+
+# Expected wavelet for indices 5000:5004 using clear_sky_index above (Matlab)
+expect_wavelet = np.array([[-0.025, 0.05, 0., -0.05, 0.025],
+                           [0.025, 0., 0., 0., -0.025],
+                           [0., 0., 0., 0., 0.]])
+
+# Expected smoothed clear sky index for indices 5000:5004 (Matlab)
+expect_cs_smooth = np.array([1., 1.0289, 1., 0.9711, 1.])
+
+
+def test_latlon_to_xy_zero():
+    coord = [0, 0]
+    pos_e = [0, 0]
+    pos = scaling.latlon_to_xy(coord)
+    assert_almost_equal(pos, pos_e, decimal=1)
+
+
+def test_latlon_to_xy_single():
+    # Must test against central value, because latlon_to_dist uses the mean
+    coord = (lat[1], lon[1])
+    pos = scaling.latlon_to_xy(coord)
+    assert_almost_equal(pos, (expect_xpos[1], expect_ypos[1]), decimal=1)
```

Comment:
```python
def test_latlon_to_xy_single(coordinates, expect_xpos, expect_ypos):
    # Must test against central value, because latlon_to_dist uses the mean
    coord = coordinates[1]  # not sure about indexing, maybe .T[1]?
    pos = scaling.latlon_to_xy(coord)
    assert_almost_equal(pos, (expect_xpos[1], expect_ypos[1]), decimal=1)
```

Reply:
consider same indexing pattern for `expect_*pos` --> `position[1]`